homeassistant/components/camera/__init__.py

```diff
@@ -290,14 +290,17 @@ def _get_camera_from_entity_id(hass: HomeAssistant, entity_id: str) -> Camera:
 def async_register_rtsp_to_web_rtc_provider(
     hass: HomeAssistant,
     domain: str,
-    provider: Callable[[str, str], Awaitable[str | None]],
+    provider: Callable[[str, str, str], Awaitable[str | None]],
```

Comment:
May want to consider making this a constant. It will make it easier for future updates.

Reply:
I've pulled it out into a type `RtspToWebRtcProviderType`

Reply:
Looks like python 3.8 doesn't like this

Reply:
If TYPE_CHECKING? It is not a big deal if you cannot get it to work though, just a suggestion

Reply:
I got it to work by going back to the old `typing` versions of these classes.